glean-core/android/src/test/java/mozilla/telemetry/glean/net/BaseUploaderTest.kt

```diff
@@ -42,61 +38,10 @@ class BaseUploaderTest {
         val uploader = spy<BaseUploader>(BaseUploader(TestUploader()))
 
         val expectedUrl = testDefaultConfig.serverEndpoint + testPath
-        uploader.doUpload(testPath, testPing, testDefaultConfig)
+        uploader.doUpload(testPath, testPing, testHeaders, testDefaultConfig)
         verify(uploader).upload(eq(expectedUrl), any(), any())
     }
 
-    @Test
-    fun `All headers are correctly reported for upload`() {
-        val uploader = spy<BaseUploader>(BaseUploader(TestUploader()))
-        `when`(uploader.getCalendarInstance()).thenAnswer {
-            val fakeNow = Calendar.getInstance()
-            fakeNow.clear()
-            fakeNow.timeZone = TimeZone.getTimeZone("GMT")
-            fakeNow.set(2015, 6, 11, 11, 0, 0)
-            fakeNow
-        }
-
-        uploader.doUpload(testPath, testPing, testDefaultConfig)
-        val headersCaptor = argumentCaptor<HeadersList>()
-
-        val expectedUrl = testDefaultConfig.serverEndpoint + testPath
-        verify(uploader).upload(eq(expectedUrl), eq(testPing), headersCaptor.capture())
-
-        val expectedHeaders = mapOf(
-            "Content-Type" to "application/json; charset=utf-8",
-            "Date" to "Sat, 11 Jul 2015 11:00:00 GMT",
-            "User-Agent" to "Glean/Test 25.0.2",
-            "X-Client-Type" to "Glean",
-            "X-Client-Version" to BuildConfig.LIBRARY_VERSION
-        )
-
-        expectedHeaders.forEach { (headerName, headerValue) ->
-            assertEquals(
-                headerValue,
-                headersCaptor.value.find { it.first == headerName }!!.second
-            )
-        }
-    }
-
-    @Test
-    fun `X-Debug-ID header is correctly added when pingTag is not null`() {
```

Comment:
Can't we keep this or do we have another test ensuring this?

Reply:
Hum, yes we can keep it. It doesn't make sense to be in this suite. I will add a tests for this behaviour in GleanTest.kt.

Reply:
Let's not move this in this PR, please. Let's keep only the changes related making upload work :)

Reply:
Oh, I am so sorry, I didn't see this comment on time and already sent it in. It is not a major test and also it's the only one I kinda re-wrote for this PR, so I guess it's fine? Let me know.